File: 0060-permutation-sequence/0060-permutation-sequence.cpp
class Solution {
    public:
        void perm(int ind,vector<int> &v,vector<vector<int>> &res,int k){
            if(res.size()==k+35000){
                return;
            }
            if(ind==v.size()){
                res.push_back(v);
                return;
            }
            for(int i=ind;i<v.size();i++){
                swap(v[i],v[ind]);
                perm(ind+1,v,res,k);
                swap(v[i],v[ind]);
            }
        }
public:
    string getPermutation(int n, int k) {
        vector<int>v;
        vector<vector<int>> res;
        for(int i=1;i<=n;i++){
            v.push_back(i);
        }
        perm(0,v,res,k);
        sort(res.begin(),res.end());
        string s="";
        for(int i:res[k-1]){
            s+=to_string(i);
        }
        return s;
    }
};

File: 0060-permutation-sequence/0060-permutation-sequence.py
class Solution:
    def perm(self,ind,li,res,k):
        if len(res)==k+35000:
            return
        if ind==len(li):
            res.append(list(li))
            return
        for i in range(ind,len(li)):
            li[i],li[ind]=li[ind],li[i]
            self.perm(ind+1,li,res,k)
            li[i],li[ind]=li[ind],li[i]
    def getPermutation(self, n: int, k: int) -> str:
        li=[]
        res=[]
        for i in range(1,n+1):
            li.append(i)
        self.perm(0,li,res,k)
        res.sort()
        ans=""
        for i in res[k-1]:
            ans+=str(i)
        return ans